File: frontendreact/movie_page/src/model/movie.ts
import { ReadCommentModel } from "./comment";
import { DirectorModel } from "./director";
import { GenreModel } from "./genre";
import { UserModel } from "./user";

export interface MovieModel {
    id: number | null,
    title: string | null,
    premiereDate: string | null,
    description: string | null,
    location: string | null,
    user: UserModel | null,
    genre: GenreModel | null,
    comments: Array<ReadCommentModel> | null,
    directors: Array<DirectorModel> | null,
}

export interface SimpleMovieModel {
    id: number | null,
    title: string | null,
    premiereDate: string | null,
    description: string | null,
    location: string | null,
}

export interface CreateMovieModel {
    title: string | null,
    premiereDate: string | null,
    description: string | null,
    location: string | null,
    user: UserModel | null,
    genre: GenreModel | null,
    directors: Array<DirectorModel> | null,
}

File: api/models/movie.go
package models

import (
	"gorm.io/gorm"
)

type Movie struct {
	gorm.Model
	Title        string `gorm:"type:varchar(100)"`
	PremiereDate string `gorm:"type:varchar(100)"`
	Description  string `gorm:"type:varchar(355)"`
	Location     string `gorm:"type:varchar(75)"`
	Poster       []byte
	Teaser       string `grom:"type:varchar(512)"`
	UserId       uint
	User         User `gorm:"foreignKey:UserId"`
	GenreId      uint
	Genre        Genre `gorm:"foreignKey:GenreId"`
	Comments     []Comment
	Directors    []Director `gorm:"many2many:movies_directors;"`
}
